.ado/jobs/test-react-native-macos-init.yml

```diff
@@ -83,6 +83,14 @@ jobs:
             npx react-native-macos-init --version latest --overwrite --prerelease
```

Comment:
@Saadnajmi: This CI only tests latest _published_ version? Shouldn't this be testing `main`?

Reply:
I think it publishes to verdacio first?

Reply:
Yep, see line 34. It publishes the local version to a verdaccio server and then tests that.

Reply:
It's clearly not using it:
![image](https://github.com/user-attachments/assets/fde52178-7d8d-4649-a278-ae239d573842)

This is what the line should look like: https://github.com/microsoft/react-native-macos/blob/8ea9b6a5a9df035ee7324a2d639e7dd1118a8519/packages/react-native/react-native.config.js#L45

The line number is also wrong.

I can't see what we're using the Verdaccio server at all. Shouldn't there be an `.npmrc` file that points to it somewhere?

Reply:
Well, it's no wonder. The publish step doesn't actually do anything. I guess this PR is now about fixing the pipeline…